internal/runners/initialize/init.go

```diff
@@ -111,79 +115,62 @@ func (r *Initialize) Run(params *RunParams) error {
 }
 
 func run(params *RunParams, out output.Outputer, proj *project.Project) (string, error) {
-	// try to initialize project at --path if provided
-	if params.Path != "" {
-		var fail *failures.Failure
-		proj, fail = project.Parse(filepath.Join(params.Path, constants.ConfigFileName))
-		if fail != nil {
-			logging.Debug("Could not parse project file at path: %v", fail.ToError())
-			proj = nil
-		}
+	if err := sanitizePath(params); err != nil {
+		return "", locale.WrapInputError(err, "err_init_sanitize_path", "Could not prepare path: {{.V0}}", err.Error())
 	}
 
-	var path string
-	// check if we are converting a headless commit
-	convertHeadless := proj != nil && proj.IsHeadless()
-	if convertHeadless {
-		if params.Language != "" {
-			return "", locale.NewInputError("init_headless_lang_provided_err", "Language argument cannot be provided when converting a headless commit.")
-		}
-	} else {
-		if err := prepare(params); err != nil {
-			return "", err
+	proj, fail := project.FromPath(params.Path)
+	if fail != nil {
+		if !projectfile.FailNoProject.Matches(fail.Type) {
+			return "", locale.WrapError(fail, "err_init_project", "Could not parse project information.")
 		}
-		path = params.Path
+		proj = nil
+	}
 
-		logging.Debug("Init: %s/%s %v", params.Namespace.Owner, params.Namespace.Project, params.Private)
+	isHeadless := proj != nil && proj.IsHeadless()
 
-		createParams := &projectfile.CreateParams{
-			Owner:           params.Namespace.Owner,
-			Project:         params.Namespace.Project,
-			Language:        params.language.String(),
-			LanguageVersion: params.version,
-			Directory:       path,
-			Private:         params.Private,
-		}
-
-		if params.Style == SkeletonEditor {
-			box := packr.NewBox("../../../assets/")
-			createParams.Content = box.String("activestate.yaml.editor.tpl")
-		}
+	// Sanitize rest of params
+	if err := sanitize(params, isHeadless); err != nil {
+		return "", err
+	}
 
-		fail := projectfile.Create(createParams)
-		if fail != nil {
-			return "", fail
-		}
+	_, err := fileutils.PrepareDir(params.Path)
+	if err != nil {
+		return "", locale.WrapError(err, "err_init_preparedir", "Could not create directory at [NOTICE]{{.V0}}[/RESET]. Error: {{.V1}}", params.Path, err.Error())
 	}
 
-	if convertHeadless {
-		headlessCommitID := proj.CommitID()
+	logging.Debug("Init: %s/%s %v", params.Namespace.Owner, params.Namespace.Project, params.Private)
 
-		err := proj.Source().SetNamespace(params.Namespace.Owner, params.Namespace.Project)
-		if err != nil {
-			return "", errs.Wrap(err, "Could not set namespace in project file.")
-		}
+	createParams := &projectfile.CreateParams{
```

Comment:
Interesting!  But do we really want to replace the entire `activestate.yaml` with a vanilla one at this point?  The user might have added changes during the headless edits that they want to keep...

Reply:
I think you were looking at an earlier commit. Please refresh.